src/field_theory/minpoly/gcd_monoid.lean

```diff
@@ -87,47 +135,91 @@ begin
   rw [aeval_map_algebra_map, aeval_algebra_map_apply, aeval_prim_part_eq_zero hP hroot, map_zero]
 end
 
+variable [is_integrally_closed R]
+
+/-- For integrally closed rings, the minimal polynomial divides any polynomial that has the
+  integral element as root. See also `minpoly.dvd` which relaxes the assumptions on `S`
+  in exchange for stronger assumptions on `R`. -/
+theorem is_integrally_closed_dvd [nontrivial R] (p : R[X]) {s : S} (hs : is_integral R s) :
```

Comment:
Sometimes the name is `is_integrally_closed_`, sometimes `is_integrally_closed.`. We should have just one convention.